opsdroid/connector/twitch/__init__.py

```diff
@@ -0,0 +1,576 @@
+import os
+import re
+import logging
+import asyncio
+import aiohttp
+import websockets
+import json
+
+from voluptuous import Required
+
+import opsdroid.connector.twitch.events as twitch_event
+
+from opsdroid.connector import Connector, register_event
+from opsdroid.events import Message
+from opsdroid.const import TWITCH_API_ENDPOINT, TWITCH_WEBHOOK_ENDPOINT, DEFAULT_ROOT_PATH
+
+
+CONFIG_SCHEMA = {
+    Required("code"): str,
+    Required("client-id"): str,
+    Required("client-secret"): str,
+    Required("channel"): str,
+    "redirect": str,
+    "forward-url": str
+}
+
+
+_LOGGER = logging.getLogger(__name__)
+TWITCH_JSON = os.path.join(DEFAULT_ROOT_PATH, 'twitch.json')
+
+
+def build_api_url(client_id, client_secret, authorization, redirect):
+    """Build api url to authenticate and refresh token."""
+    clientid = f"?client_id={client_id}"
+    secret = f"&client_secret={client_secret}"
+    grant = f"&grant_type={authorization}"
+    redirect = f"&redirect_uri={redirect}"
+
+    return f"{TWITCH_API_ENDPOINT}{clientid}{secret}{grant}{redirect}"
```

Comment:
Have you seen the [docs on passing URL paramaters with aiohttp](https://docs.aiohttp.org/en/stable/client_quickstart.html#passing-parameters-in-urls)? You don't need to build urls like this, you can pass a dictionary of parameters in the `get` method and it will do it for you.